Jakefile.js

```diff
@@ -755,13 +755,18 @@ function runConsoleTests(defaultReporter, runInParallel) {
             // schedule work for chunks
             configFiles.forEach(function (f) {
                 var configPath = path.join(taskConfigsFolder, f);
-                var workerCmd = "mocha" + " -t " + testTimeout + " -R " + reporter + " " + colors + " " + run + " --config='" + configPath + "'";
+                var workerCmd = "mocha" + " -t " + testTimeout + " -R " + reporter + " " + colors + bail + " " + run + " --config='" + configPath + "'";
                 console.log(workerCmd);
                 exec(workerCmd,  finishWorker, finishWorker) 
             });
             
             function finishWorker(e, errorStatus) {
                 counter--;
+
+                if (bail && errorStatus !== undefined) {
+                    failWithStatus(firstErrorStatus);

```

Comment:
Does this fail with the correct number of errors? I don't think `firstErrorStatus` would be assigned until after the next `if` block.

I guess the fix would be to move below the next `if` block or just return `errorStatus`